packages/core-bridge/src/worker.rs

```diff
@@ -453,14 +461,44 @@ mod config {
         },
     }
 
+    #[derive(TryFromJs)]
+    pub struct WorkerDeploymentOptions {
+        version: WorkerDeploymentVersion,
+        use_worker_versioning: bool,
+        default_versioning_behavior: VersioningBehavior,
+    }
+
+    #[derive(TryFromJs)]
+    pub struct WorkerDeploymentVersion {
+        build_id: String,
+        deployment_name: String,
+    }
+
+    #[derive(TryFromJs)]
+    pub enum VersioningBehavior {
+        Pinned,
+        AutoUpgrade,
+    }
+
     impl BridgeWorkerOptions {
         pub(crate) fn into_core_config(self) -> Result<WorkerConfig, WorkerConfigBuilderError> {
             // Set all other options
             let mut builder = WorkerConfigBuilder::default();
             builder
                 .client_identity_override(Some(self.identity))
-                .worker_build_id(self.build_id)
-                .use_worker_versioning(self.use_versioning)
+                .versioning_strategy({
```

Comment:
I suppose you may have written that part before the refactor…

Going forward, I really want to avoid having the bridge perform these kind of X-to-Y mappings, fallback/legacy/default value handling, etc. That kind of logic is easier to deal in lang, and that result in code that is much more easier to review. In this case, what I would do is to have bridge define an enum that replicates Core's `WorkerVersioningStrategy` enum, with the same variants and variant properties.

But feel free to disregard for now, I can revisit at a later point.